### PR TITLE
Chen's work on personal API

### PR DESCRIPTION
**Pull request by achentha**
Seeding was working for awhile. But later on it does not quite update in heroku any more.

### CODE REVIEW

File: views/index.html [hunk 1]
```diff
@@ -4,26 +4,113 @@
...
-    <script src="//ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.0/js/bootstrap.min.js"></script>
 
     <!-- APPLICATION SCRIPTS -->
-    <script src="/scripts/app.js"></script>
+    <script src="scripts/app.js"></script>
   </head>
   <body>
+
```
Comment (breese8009) [inline views/index.html:21]: Watch for indentation in your code

File: seed.js [hunk 2]
```diff
@@ -13,3 +13,98 @@
 //   console.log("Created new campsite", campsite._id)
 //   process.exit(); // we're all done! Exit the program.
 // })
+const db = require("./models");
```
Comment (breese8009) [inline seed.js:16]: indentation needed a bit more here

File: server.js [hunk 3]
```diff
@@ -37,24 +38,43 @@ app.get('/', function homepage(req, res) {
...
+    documentationUrl: "https://github.com/achentha/express-personal-api/blob/master/README.md",
+    baseUrl: "https://dry-scrubland-21249.herokuapp.com/",
     endpoints: [
       {method: "GET", path: "/api", description: "Describes all available endpoints"},
-      {method: "GET", path: "/api/profile", description: "Data about me"}, // CHANGE ME
-      {method: "POST", path: "/api/campsites", description: "E.g. Create a new campsite"} // CHANGE ME
+      {method: "GET", path: "/api/profile", description: "About me"},
+      {method: "GET", path: "/api/birds", description: "Show all my favorite birds"},
+      {method: "POST", path: "/api/birds", description: "Add a new favorite bird"},
+      {method: "PUT", path: "/api/birds", description: "Modify a favorite bird"},
```
Comment (mnfmnfm) [inline server.js]: These routes are actually (correctly!) at /api/birds/:id, not at /api/birds as this documentation says.

File: views/index.html [hunk 4]
```diff
@@ -4,26 +4,113 @@
...
 
     <!-- STYLESHEETS -->
-    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
-    <link rel="stylesheet" href="/styles/styles.css">
+    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
+    <link rel="stylesheet" href="styles/styles.css">
 
     <!-- VENDOR SCRIPTS -->
-    <script src="//ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
+    <script src="http://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script>
```
Comment (mnfmnfm) [inline views/index.html]: The fact that you load this over http always, rather than using https, means that your site doesn't work the first time on Heroku, and the user has to manually click "allow unsecure scripts".

File: seed.js (hunk 2 above)
Comment (mnfmnfm) [reply seed.js:16]: +1

File: seed.js [hunk 5]
```diff
@@ -13,3 +13,100 @@
...
+  db.Bird.create(birdSeed, function(err, birds) {
+    if (err) {
+      console.log(`failed to create birdSeed: err = ${err}`);
+      return;
+    }
+
+    console.log("successfully created birdSeed");
+  });
+
+  process.exit();  //so it does not hang there
```
Comment (mnfmnfm) [inline seed.js]: Having this at the end here is dangerous! It means that you might exit the process before the data is finished saving. This line should go INSIDE your callback (i.e. line 108/109), so that you only exit the process after the data is saved.

File: public/styles/styles.css [hunk 6]
```diff
@@ -1,10 +1,77 @@
...
 body {
   color: #333;
   font-family: Helvetica, Arial, sans-serif;
-  background-color: skyblue; /* Sanity Check! */
+  background-image: url("../images/background12.jpeg");
+  /*background-repeat: no-repeat;*/
+  background-size: contain;
+}
+
+#bird-form {
```
Comment (mnfmnfm) [inline public/styles/styles.css]: empty styling feels like a leftover.

File: public/scripts/app.js [hunk 7]
```diff
@@ -2,6 +2,171 @@ console.log("Sanity Check: JS is working!");
...
+
+    $("[data-bird-id=" + addedBird._id + "]")[0].scrollIntoView();
+  });
+  $(this).trigger("reset");
+}
+
+function handelEditBird(e) {
+  e.preventDefault();
+  let birdId = $(this).closest(".bird").data("bird-id");
+  let $bird = $(this).closest(".bird");
```
Comment (mnfmnfm) [inline public/scripts/app.js:99]: These two lines do the same bunch of work, and you could put them in the opposite order to great effect!

```js
let $bird = $(this).closest(".bird");
let birdId = $bird.data("bird-id");
```

File: public/scripts/app.js [hunk 8]
```diff
@@ -2,6 +2,171 @@ console.log("Sanity Check: JS is working!");
...
+  console.log(`handelEditBird: birdId = ${birdId}`);
+
+  //toggle the save & edit button
+  $bird.find(".save-bird").toggleClass("hidden");
+  $bird.find(".edit-bird").toggleClass("hidden");
+
+  //change the bird name in the span to become input with the value of the current bird name
+  let birdName = $bird.find(".bird-name").text();
+  $bird.find(".bird-name").html(`<input class="edit-bird-name" value="${birdName}"></input>`);
```
Comment (mnfmnfm) [inline public/scripts/app.js:108]: similarly, could do less work here:
```js
let $birdName = $bird.find(".bird-name");
let oldName = $birdName.text();
$birdName.html(`...`)
```

File: public/scripts/app.js [hunk 9]
```diff
@@ -2,6 +2,171 @@ console.log("Sanity Check: JS is working!");
...
+}
+
+function handleBirdUpdate(updatedBird) {
+  let birdId = updatedBird._id;
+
+  //remove old bird item
+  $("[data-bird-id=" + birdId +"]").remove();
+  renderBird(updatedBird);
+
+  $("[data-bird-id=" + birdId + "]")[0].scrollIntoView();
```
Comment (mnfmnfm) [inline public/scripts/app.js:152]: awesome use of this method!

File: public/scripts/app.js [hunk 10]
```diff
@@ -2,6 +2,171 @@ console.log("Sanity Check: JS is working!");
...
+//adding a bird
+$("#bird-form form").on("submit", handleAddBird);
+//handling edit button click for editing a bird
+$("#birds").on("click", ".edit-bird", handelEditBird);
+//handling save button click for editing a bird
+$("#birds").on("click", ".save-bird", handleSaveBird);
+//handling delete button of a bird
+$("#birds").on("click", ".delete-bird", handleDeleteBird);
+
+function renderAllBirds(birds) {
```
Comment (mnfmnfm) [inline public/scripts/app.js:24]: this is lovely.

File: controllers/birdsController.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    console.log("create bird success");
+    res.json(bird);
+  });
+}
+
+function update(req, res) {
+  db.Bird.findById(req.params.birdId, function(err, bird){
+    if (err) {
+      console.log(`Did not find bird id: ${req.params.birdId} in db`);
+      return;
```
Comment (mnfmnfm) [inline controllers/birdsController.js:32]: You should probably send a response here, not just return from the function; as is, your app.js file would be waiting forever, for a response that would never come, if there were an error. Instead, you could send a 404.

Comment (mnfmnfm) [reply controllers/birdsController.js:32]: And this goes for all of your error-handling: you should send a response, not just console.log.

**Review by mnfmnfm — COMMENTED** [review]
Overall, this looks really good. You made it through all the requirements and have reasonable styling. Nice work.
